build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -94,38 +100,79 @@ class AssetGraph {
   /// set of assets that need to be deleted.
   Iterable<AssetId> updateAndInvalidate(Map<AssetId, ChangeType> updates) {
     var deletes = new Set<AssetId>();
-    var seen = new Set<AssetId>();
     void clearNodeAndDeps(AssetId id, ChangeType rootChangeType,
-        {AssetId parent}) {
-      if (seen.contains(id)) return;
-      seen.add(id);
+        {AssetId parent, bool rootIsSource}) {
```

Comment:
Oh, actually that makes sense. If somebody deletes a generated output then we should regenerate it. It is a bit of a weird case but I think it makes sense to support.